### PR TITLE
feat: sepolia flag and config

### PR DESCRIPTION
**Pull request by vyzo**
## 1. Purpose or design rationale of this PR

Adds a `--scroll-sepolia` flag and the necessary configuration to run a sepolia l2geth node.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [X] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [X] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [X] This PR is not a breaking change
- [ ] Yes

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204964693191470

### CODE REVIEW

File: params/version.go [hunk 1]
```diff
@@ -44,7 +44,8 @@ var VersionWithMeta = func() string {
 
 // ArchiveVersion holds the textual version string used for Geth archives.
 // e.g. "1.8.11-dea1ce05" for stable releases, or
-//      "1.8.13-unstable-21c059b6" for unstable releases
+//
```
Comment (Thegaram) [inline params/version.go]:
```suggestion
```

File: params/config.go [hunk 2]
```diff
@@ -296,6 +297,44 @@ var (
...
+		EIP158Block:         big.NewInt(0),
+		ByzantiumBlock:      big.NewInt(0),
+		ConstantinopleBlock: big.NewInt(0),
+		PetersburgBlock:     big.NewInt(0),
+		IstanbulBlock:       big.NewInt(0),
+		MuirGlacierBlock:    nil,
+		BerlinBlock:         big.NewInt(0),
+		LondonBlock:         big.NewInt(0),
+		ArrowGlacierBlock:   nil,
+		ArchimedesBlock:     big.NewInt(2646311),
```
Comment (Thegaram) [inline params/config.go]:
```suggestion
		ArchimedesBlock:     big.NewInt(0),
```

File: params/config.go [hunk 3]
```diff
@@ -296,6 +297,44 @@ var (
...
+		ShanghaiBlock:       nil,
```
Comment (Thegaram) [inline params/config.go]:
```suggestion
		ShanghaiBlock:       big.NewInt(0),
```

File: core/genesis.go [hunk 4]
```diff
@@ -144,10 +144,10 @@ func (e *GenesisMismatchError) Error() string {
...
 // The block that will be used is:
 //
-//                          genesis == nil       genesis != nil
-//                       +------------------------------------------
-//     db has no genesis |  main-net default  |  genesis
-//     db has genesis    |  from DB           |  genesis (if compatible)
+//	                     genesis == nil       genesis != nil
+//	                  +------------------------------------------
+//	db has no genesis |  main-net default  |  genesis
+//	db has genesis    |  from DB           |  genesis (if compatible)
```
Comment (Thegaram) [inline core/genesis.go:150]: Where these changed by your IDE?

File: cmd/utils/flags.go [hunk 5]
```diff
@@ -1705,6 +1717,11 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		}
 		cfg.Genesis = core.DefaultScrollAlphaGenesisBlock()
 		// SetDNSDiscoveryDefaults(cfg, params.ScrollAlphaGenesisHash)
+	case ctx.GlobalBool(ScrollSepoliaFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 534351
```
Comment (Thegaram) [inline cmd/utils/flags.go:1735]: Can we explicitly set `l1.confirmations` and `l1.sync.startblock` here?

File: core/genesis.go (hunk 4 above)
Comment (vyzo) [reply core/genesis.go:150]: gofmt does it... i have it set in my save hook in emacs.

File: cmd/utils/flags.go (hunk 5 above)
Comment (vyzo) [reply cmd/utils/flags.go:1735]: sure, will do.

Comment (vyzo) [reply cmd/utils/flags.go:1735]: done -- https://github.com/scroll-tech/go-ethereum/pull/377/commits/fe1bbba6c477ca1d1b1336e4b81507b72e11d6b2

**Review by Thegaram — COMMENTED** [review]
LGTM, let's update, test, and merge once Sepolia is running.

**Review by Thegaram — APPROVED** [review]

**Pull request merged**